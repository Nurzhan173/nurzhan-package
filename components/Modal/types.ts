export type Modal = {
  modalElement: HTMLElement,
  closeModalBtn: HTMLElement,
  title: string,
  content: string
}
