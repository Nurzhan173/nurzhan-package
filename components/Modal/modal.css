
/* The Modal (background) */
.modal {
  display: none; /* Hidden by default */
  position: fixed; /* Stay in place */
  z-index: 1; /* Sit on top */
  padding-top: 100px; /* Location of the box */
  left: 0;
  top: 0;
  width: 100%; /* Full width */
  height: 100%; /* Full height */
  overflow: auto; /* Enable scroll if needed */
  background-color: rgb(0,0,0); /* Fallback color */
  background-color: rgba(0,0,0,0.4); /* Black w/ opacity */
}

/* Modal Content */
.modal-content {
  display: flex;
  background-color: #fefefe;
  margin: auto;
  padding: 20px;
  border: 1px solid #888;
  width: 80%;
  border-radius: 8px;
}

.modal-wrapper {
  width: 80%;
  margin: auto;
}

/* The Close Button */
.close {
  color: #aaaaaa;
  float: right;
  font-size: 28px;
  font-weight: bold;
  position: absolute;
  right: 100px;
}

.close:hover,
.close:focus {
  color: #000;
  text-decoration: none;
  cursor: pointer;
}

.openBtn {
  cursor: pointer;
}
