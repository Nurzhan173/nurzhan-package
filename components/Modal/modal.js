class Modal {
  constructor() {
    function closeModal() {
      modal.style.display = 'none';
    }

    window.onclick = function(event) {
      if (event.target === modal) {
        window.modal.style.display = 'none';
      }
    }

    document.addEventListener("keydown", function (e) {
      if (e.key === "Escape") {
        modal.style.display = 'none';
      }
    });
  }

  render() {
    const modal = document.createElement("div");
    const closeButton = document.createElement("span");
    const modalContent = document.createElement("div");

    modal.id = "modal";
    modal.className = "modal";

    closeButton.className="close";
    closeButton.textContent = "x";
    closeButton.addEventListener("click", () => {
      window.modal.style.display = 'none';
    })

    modalContent.id = "modal-content";
    modalContent.className = "modal-content";

    modal.appendChild(closeButton);
    modal.appendChild(modalContent);

    return modal;
  }
}

module.exports = Modal;
