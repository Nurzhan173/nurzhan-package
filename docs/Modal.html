<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8">
    <title>JSDoc: Class: Modal</title>

    <script src="scripts/prettify/prettify.js"> </script>
    <script src="scripts/prettify/lang-css.js"> </script>
    <!--[if lt IE 9]>
      <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
    <link type="text/css" rel="stylesheet" href="styles/prettify-tomorrow.css">
    <link type="text/css" rel="stylesheet" href="styles/jsdoc-default.css">
</head>

<body>

<div id="main">

    <h1 class="page-title">Class: Modal</h1>

    




<section>

<header>
    
        <h2><span class="attribs"><span class="type-signature"></span></span>Modal<span class="signature">()</span><span class="type-signature"></span></h2>
        
    
</header>

<article>
    <div class="container-overview">
    
        

    

    
    <h4 class="name" id="Modal"><span class="type-signature"></span>new Modal<span class="signature">()</span><span class="type-signature"></span></h4>
    

    















<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="modal.ts.html">modal.ts</a>, <a href="modal.ts.html#line1">line 1</a>
    </li></ul></dd>
    

    

    

    
</dl>




















    
    </div>

    

    

    

    

    

    

    
        <h3 class="subsection-title">Members</h3>

        
            
<h4 class="name" id="closeModalBtn"><span class="type-signature"></span>closeModalBtn<span class="type-signature"> :HTMLElement</span></h4>




<div class="description">
    <p>close button</p>
</div>



    <h5>Type:</h5>
    <ul>
        <li>
            
<span class="param-type">HTMLElement</span>


        </li>
    </ul>





<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="modal.ts.html">modal.ts</a>, <a href="modal.ts.html#line43">line 43</a>
    </li></ul></dd>
    

    

    

    
</dl>






        
            
<h4 class="name" id="content"><span class="type-signature"></span>content<span class="type-signature"> :string</span></h4>




<div class="description">
    <p>content</p>
</div>



    <h5>Type:</h5>
    <ul>
        <li>
            
<span class="param-type">string</span>


        </li>
    </ul>





<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="modal.ts.html">modal.ts</a>, <a href="modal.ts.html#line31">line 31</a>
    </li></ul></dd>
    

    

    

    
</dl>






        
            
<h4 class="name" id="modalElement"><span class="type-signature"></span>modalElement<span class="type-signature"> :HTMLElement</span></h4>




<div class="description">
    <p>modal wrapper</p>
</div>



    <h5>Type:</h5>
    <ul>
        <li>
            
<span class="param-type">HTMLElement</span>


        </li>
    </ul>





<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="modal.ts.html">modal.ts</a>, <a href="modal.ts.html#line37">line 37</a>
    </li></ul></dd>
    

    

    

    
</dl>






        
            
<h4 class="name" id="title"><span class="type-signature"></span>title<span class="type-signature"> :string</span></h4>




<div class="description">
    <p>title</p>
</div>



    <h5>Type:</h5>
    <ul>
        <li>
            
<span class="param-type">string</span>


        </li>
    </ul>





<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="modal.ts.html">modal.ts</a>, <a href="modal.ts.html#line25">line 25</a>
    </li></ul></dd>
    

    

    

    
</dl>






        
    

    

    

    
</article>

</section>







<section>

<header>
    
        <h2><span class="attribs"><span class="type-signature"></span></span>Modal<span class="signature">(onclick, b)</span><span class="type-signature"> &rarr; {number}</span></h2>
        
            <div class="class-description"><p>Функция, складывающая два числа.</p></div>
        
    
</header>

<article>
    <div class="container-overview">
    
        

    
    <h2>Constructor</h2>
    

    
    <h4 class="name" id="Modal"><span class="type-signature"></span>new Modal<span class="signature">(onclick, b)</span><span class="type-signature"> &rarr; {number}</span></h4>
    

    











    <h5>Parameters:</h5>
    

<table class="params">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>onclick</code></td>
            

            <td class="type">
            
                
<span class="param-type">function</span>


            
            </td>

            

            

            <td class="description last"><p>onClick</p></td>
        </tr>

    

        <tr>
            
                <td class="name"><code>b</code></td>
            

            <td class="type">
            
                
<span class="param-type">number</span>


            
            </td>

            

            

            <td class="description last"><p>Второе число.</p></td>
        </tr>

    
    </tbody>
</table>






<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="modal.ts.html">modal.ts</a>, <a href="modal.ts.html#line15">line 15</a>
    </li></ul></dd>
    

    

    

    
</dl>















<h5>Returns:</h5>

        
<div class="param-desc">
    <p>Сумма двух чисел.</p>
</div>



<dl>
    <dt>
        Type
    </dt>
    <dd>
        
<span class="param-type">number</span>


    </dd>
</dl>

    





    
    </div>

    

    

    

    

    

    

    
        <h3 class="subsection-title">Members</h3>

        
            
<h4 class="name" id="closeModalBtn"><span class="type-signature"></span>closeModalBtn<span class="type-signature"> :HTMLElement</span></h4>




<div class="description">
    <p>close button</p>
</div>



    <h5>Type:</h5>
    <ul>
        <li>
            
<span class="param-type">HTMLElement</span>


        </li>
    </ul>





<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="modal.ts.html">modal.ts</a>, <a href="modal.ts.html#line43">line 43</a>
    </li></ul></dd>
    

    

    

    
</dl>






        
            
<h4 class="name" id="content"><span class="type-signature"></span>content<span class="type-signature"> :string</span></h4>




<div class="description">
    <p>content</p>
</div>



    <h5>Type:</h5>
    <ul>
        <li>
            
<span class="param-type">string</span>


        </li>
    </ul>





<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="modal.ts.html">modal.ts</a>, <a href="modal.ts.html#line31">line 31</a>
    </li></ul></dd>
    

    

    

    
</dl>






        
            
<h4 class="name" id="modalElement"><span class="type-signature"></span>modalElement<span class="type-signature"> :HTMLElement</span></h4>




<div class="description">
    <p>modal wrapper</p>
</div>



    <h5>Type:</h5>
    <ul>
        <li>
            
<span class="param-type">HTMLElement</span>


        </li>
    </ul>





<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="modal.ts.html">modal.ts</a>, <a href="modal.ts.html#line37">line 37</a>
    </li></ul></dd>
    

    

    

    
</dl>






        
            
<h4 class="name" id="title"><span class="type-signature"></span>title<span class="type-signature"> :string</span></h4>




<div class="description">
    <p>title</p>
</div>



    <h5>Type:</h5>
    <ul>
        <li>
            
<span class="param-type">string</span>


        </li>
    </ul>





<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="modal.ts.html">modal.ts</a>, <a href="modal.ts.html#line25">line 25</a>
    </li></ul></dd>
    

    

    

    
</dl>






        
    

    

    

    
</article>

</section>




</div>

<nav>
    <h2><a href="index.html">Home</a></h2><h3>Classes</h3><ul><li><a href="Modal.html">Modal</a></li></ul>
</nav>

<br class="clear">

<footer>
    Documentation generated by <a href="https://github.com/jsdoc/jsdoc">JSDoc 4.0.2</a> on Wed Sep 20 2023 23:37:34 GMT+0600 (East Kazakhstan Time)
</footer>

<script> prettyPrint(); </script>
<script src="scripts/linenumber.js"> </script>
</body>
</html>