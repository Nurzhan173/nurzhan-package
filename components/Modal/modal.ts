import './modal.css';

/**
 * @module Modal
 */

/**
 * Component Modal.
 *
 * @param {{title: string, content: string}} modal
 */


class Modal {

  /**
   * title
   * @type {string}
   */
  title;

  /**
   * content
   * @type {string}
   */
  content;

  /**
   * modal wrapper
   * @type {HTMLElement}
   */
  modalElement;

  /**
   * close button
   * @type {HTMLElement}
   */
  closeModalBtn;


  constructor(title, content) {
    this.title = title;
    this.content = content;

    const titleElement = document.createElement('h2');
    titleElement.innerText = this.title;

    const contentElement = document.createElement('p');
    contentElement.innerText = this.content;

    this.modalElement = document.createElement('div');
    this.modalElement.className = 'modal';
    this.modalElement.style.display = 'none';

    this.closeModalBtn = document.createElement('span');
    this.closeModalBtn.className = 'close-btn';
    this.closeModalBtn.innerText = 'Ã—';

    const modalContent = document.createElement('div');
    modalContent.className = 'modal-content';
    modalContent.appendChild(titleElement);
    modalContent.appendChild(contentElement);
    modalContent.appendChild(this.closeModalBtn);

    this.modalElement.appendChild(modalContent);

    document.body.appendChild(this.modalElement);

    this.closeModalBtn.addEventListener('click', () => {
      this.close();
    });

    window.addEventListener('click', (event) => {
      if (event.target === this.modalElement) {
        this.close();
      }
    });
  }

  open() {
    this.modalElement.style.display = 'block';
  }

  close() {
    this.modalElement.style.display = 'none';
  }
}

export default Modal;
