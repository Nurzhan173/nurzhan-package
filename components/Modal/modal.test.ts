import Modal from './modal';

import { fireEvent } from '@testing-library/dom';
import { describe, beforeEach, afterEach, test, expect } from '@jest/globals'

describe('Modal Component', () => {
  let modal: any;

  beforeEach(() => {
    modal = new Modal('Test Title', 'Test Content');
  });

  afterEach(() => {
    document.body.innerHTML = '';
  });

  test('modal should be hidden initially', () => {
    expect(modal).toMatchSnapshot();
    expect(modal.modalElement.style.display).toBe('none');
  });

  test('modal should show when open() is called', () => {
    modal.open();
    expect(modal).toMatchSnapshot();
    expect(modal.modalElement.style.display).toBe('block');
  });

  test('modal should hide when close() is called', () => {
    modal.open();
    modal.close();
    expect(modal.modalElement.style.display).toBe('none');
  });

  test('close button should call close() when clicked', () => {
    modal.open();

    const closeButton = modal.modalElement.querySelector('.close-btn');
    fireEvent.click(closeButton);

    expect(modal.modalElement.style.display).toBe('none');
  });

  test('modal should close when clicking outside the modal content', () => {
    modal.open();

    const modalContent = modal.modalElement.querySelector('.modal-content');
    fireEvent.click(modalContent); // Simulate click inside modal content

    expect(modal.modalElement.style.display).toBe('block');
  });
});
