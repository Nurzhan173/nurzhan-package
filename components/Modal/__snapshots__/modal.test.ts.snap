// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Component modal should be hidden initially 1`] = `
Modal {
  "closeModalBtn": <span
    class="close-btn"
  />,
  "content": "Test Content",
  "modalElement": <div
    class="modal"
    style="display: none;"
  >
    <div
      class="modal-content"
    >
      <h2 />
      <p />
      <span
        class="close-btn"
      />
    </div>
  </div>,
  "title": "Test Title",
}
`;

exports[`Modal Component modal should show when open() is called 1`] = `
Modal {
  "closeModalBtn": <span
    class="close-btn"
  />,
  "content": "Test Content",
  "modalElement": <div
    class="modal"
    style="display: block;"
  >
    <div
      class="modal-content"
    >
      <h2 />
      <p />
      <span
        class="close-btn"
      />
    </div>
  </div>,
  "title": "Test Title",
}
`;
